docs/designs/transaction-design.md

```diff
@@ -0,0 +1,222 @@
+    ItemID INT,                                         
+    TransactionType VARCHAR(50),                          
+    TotalAmount DECIMAL(10, 2),                         
+    TransactionDate DATETIME DEFAULT GETDATE(),         
+    
+    -- Define foreign key constraints
+    FOREIGN KEY (UserID) REFERENCES Users(UserID),      
+    FOREIGN KEY (ItemID) REFERENCES Items(ItemID)       
+);
+
+-- Optionally, we may want to create an index for faster searching by TransactionGUID, UserID, and TransactionDate
+CREATE INDEX IX_TransactionHistory_TransactionGUID ON TransactionHistory(TransactionGUID);
+CREATE INDEX IX_TransactionHistory_UserID ON TransactionHistory(UserID);
+CREATE INDEX IX_TransactionHistory_TransactionDate ON TransactionHistory(TransactionDate);
+```
+
+**Stored Procedures:**
+- **RecordTransaction**:  
+  - Inserts a new transaction record into the `TransactionHistory` table.
+  - Parameters: `UserID`, `TransactionGUID`, `ItemID`, `TransactionType`, `TotalAmount`.
+  
+```sql
+CREATE PROCEDURE LogTransaction
+    @UserID INT,                     
+    @TransactionGUID UNIQUEIDENTIFIER, 
+    @ItemID INT,                   
+    @TransactionType VARCHAR(50),     
+    @TotalAmount DECIMAL(10, 2)       
+AS
+BEGIN
+    -- Insert a new record into the TransactionHistory table
+    INSERT INTO TransactionHistory (
+        TransactionID,
+        UserID,
+        TransactionGUID,
+        ItemID,
+        TransactionType,
+        TotalAmount,
+        TransactionDate
+    )
+    VALUES (
+        NEWID(),                     -- Generate a new GUID for the TransactionID
+        @UserID,                     
+        @TransactionGUID,           
+        @ItemID,                   
+        @TransactionType,             
+        @TotalAmount,               
+        GETDATE()                 
+    );
+END;
+```
+
+- **GetTransactionHistory**:
+- Retrieves all transactions associated with a specific `UserID`(admin user only).
+
+```sql
+CREATE PROCEDURE GetTransactionHistory
+    @AdminUserID INT                 -- The ID of the admin requesting transaction history
+AS
+BEGIN
+    -- Check if the requesting user is an admin
+    IF EXISTS (SELECT 1 FROM Users WHERE UserID = @AdminUserID AND IsAdmin = 1)
+    BEGIN
+        -- Retrieve all transaction history records
+        SELECT 
+            th.TransactionID,
+            th.UserID,
+            th.TransactionGUID,
+            th.ItemID,
+            th.TransactionType,
+            th.TotalAmount,
+            th.TransactionDate
+        FROM 
+            TransactionHistory th
+        INNER JOIN 
+            Items i ON th.ItemID = i.ItemID
+        ORDER BY 
+            th.TransactionDate DESC;      -- Optionally order by transaction date or any other field
+    END
+    ELSE
+    BEGIN
+        -- If not an admin, return an error message
+        THROW 50000, 'Access Denied. Only admins can access transaction history.', 1;
+    END
+END;
+```
+
+### API Endpoints
```

Comment:
These are easily exposed through the json file with the other REST API endpoints.